flutter_local_notifications/example/lib/main.dart

```diff
@@ -806,7 +880,126 @@ class _HomePageState extends State<HomePage> {
     const NotificationDetails platformChannelSpecifics =
         NotificationDetails(android: androidPlatformChannelSpecifics);
     await flutterLocalNotificationsPlugin.show(
-        0, 'plain title', 'plain body', platformChannelSpecifics,
+        id++, 'plain title', 'plain body', platformChannelSpecifics,
```

Comment:
@Kavantix @ened nice idea to increment but one thing to note on why  I had originally set the id to zero in all cases was to allow for `cancel()` to dismiss the notification on the screen. It looks like the `_cancelNotification()` method still invokes logic to cancel the notification with an id of zero. This may potentially look like it's not cancelling anything if multiple notification requests are made. If changes are going to be made around the id is set then a way to cancel a notification with a specific id is required and allows users of the example to know exactly what is being cancelled/deleted

Reply:
Good point, we can make cancel also increment so they are also cancelled in order?

Reply:
Or we could even show a list of notifications that were scheduled and allow them to cancel a specific one

Reply:
Let's think who should keep track of notification IDs. 

I think it's the client that's responsible, so I would prefer to store the notification ids in a list, then let the "cancel" function in the example iterate through the list and cancel them one by one, then clear the list.

In a real world App you'd have specific notification ids as constants.

Reply:
> Or we could even show a list of notifications that were scheduled and allow them to cancel a specific one

Yes, essentially that based on what I just posted, if we want things to be that complex in the example app.

Reply:
No in a real world app they are not per se constants just known values

Reply:
> No in a real world app they are not per se constants just known values

Ok agree. ;-)

Reply:
This one's tricky as there could be arguments for decrementing instead so it cancels the last notification that was shown. A really naive approach I can think of is to continue incrementing and  have more changes done so the text within the notification will have the id. With that done, a text field can be added to users specify the ID to cancel

Reply:
But yes keeping a list of ids would be simple enough, and then the button can say `cancel all notifications`

Reply:
Ah last notification first, interesting that does make sense. We could use a stack perhaps

Reply:
`I++`,
`I--`

Maybe?

Reply:
@MaikuB with that idea, you don't know the id of a notification that is scheduled in the future right, that only works for already delivered notifications.
Adding the list should be doable IMO (it is flutter after all)

Reply:
Or we could just do cancel all and actually call the cancel all method of the package?
That's by far the simplest right

Reply:
yeah the  i++/i-- approach could work

I was going to say if a list was going to be used then note that the app already has a button that calls https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/FlutterLocalNotificationsPlugin/cancelAll.html and could result in some confusion in the button text isn't clear enough. That is, instead of cancel all, perhaps it's more accurate to say something along the lines of "cancel each (individual) notification by id". 

Note: in the original code I had for the example app, given every notification had the same ID, cancel all led to same result. Its purpose was more to highlight that there's a different API in case you're wondering why I bothered putting that there :)

Reply:
This was addressed in the latest update, please take a look.